### PR TITLE
Add NHANES reference medians

### PR DESCRIPTION
**Pull request by dchud**
Adds reference medians from NHANES and changes recentering options/behavior based on `sd.recenter`. Summary of NHANES derivation method now in README.

Key things to look for (aside from code logic!):

- Does the `sd.recenter` param description in code and the README make sense? We're overloading one parameter here, and though I came down on a decision that this was the simplest way to do it, it's nowhere near simple.
- Does the NHANES derivation summary give enough detail?
- Are the updated tests meaningful/worthwhile?

### CODE REVIEW

File: R/growth.R [hunk 1]
```diff
@@ -1902,24 +1910,62 @@ cleangrowth <- function(subjid,
...
-    sd.recenter <- data.all[exclude < 'Exclude', sd_median(param, sex, agedays, sd.orig)]
-    if (sdmedian.filename != "") {
-      write.csv(sd.recenter, sdmedian.filename, row.names = F)
+    # Use NHANES medians if the string "nhanes" is specified instead of a data.table
+    # or if sd.recenter is not specified as "derive" and N < 5000.
+    if ((is.character(sd.recenter) & tolower(sd.recenter) == "nhanes") |
+      (!(is.character(sd.recenter) & tolower(sd.recenter) == "derive") & (data.all[, .N] < 5000))) {
+      nhanes_reference_medians_path <- ifelse(
+        ref.data.path == "",
+        system.file("extdata/nhanes-reference-medians.csv", package = "growthcleanr"),
```
Comment (delosh653) [inline R/growth.R:1922]: Should probably be changed to `file.path`, for more explicit applicability across different machines.

File: README.Rmd [hunk 2]
```diff
@@ -967,10 +976,61 @@ above under [Computing BMI percentiles and Z-scores](#bmi).
 same `syngrowth` synthetic example dataset as `growthcleanr`, with results
 included.
 
+## <a name="nhanes"></a>NHANES reference medians
+
+`growthcleanr` releases up to 1.2.4 offered two options for recentering medians, either
```
Comment (delosh653) [inline README.Rmd:981]: Wondering if it might be nice here to put a link to releases/tags?

**Review by delosh653 — APPROVED** [review]
I approve (with the comments I added as well). Here are my responses to your things to focus on:

- Does the sd.recenter param description in code and the README make sense? We're overloading one parameter here, and though I came down on a decision that this was the simplest way to do it, it's nowhere near simple.
	- I agree that overloading the variable is best. It would be nice if it was broken into a bulleted sub-list (see https://stackoverflow.com/questions/9267584/when-documenting-in-roxygen-how-do-i-make-an-itemized-list-in-details); the paragraph as it stands is pretty daunting. Same goes with the with the readme. I like the bullets in NHANES reference medians section of the readme.
- Does the NHANES derivation summary give enough detail?
	- I think so. I think the only change I would make is to break up the third long paragraph between the survey smoothing and the weights closer to birth, just for readability.
- Are the updated tests meaningful/worthwhile?
	- I think they're relatively meaningful (we don't want the underlying values to change), but I think it might be good to put something more explicit in the comparison between NHANES when it's specified and NHANES when it's derived -- maybe something like expect equal when it's comparing the two datatables? Otherwise, I think that there's not much other testing to be done on that front explicitly.

File: R/growth.R (hunk 1 above)
Comment (dchud) [reply R/growth.R:1922]: Sure - I *think* we want to keep the rest of the `ifelse()` the same, e.g.:

```r
nhanes_reference_medians_path <- ifelse(
        ref.data.path == "",
        system.file(file.path("extdata", "nhanes-reference-medians.csv"), package = "growthcleanr"),
        paste(ref.data.path, "nhanes-reference-medians.csv", sep = "")
      )
```

Just want to be sure I'm not overthinking the `paste()` in the else slot. This already allows users to specify their own os-appropriate paths, and I don't think we want to change that.

File: README.Rmd (hunk 2 above)
Comment (dchud) [reply README.Rmd:981]: Good idea! Added a link under Changes section below.

**Comment by dchud** [comment]
Re: formatting the param description, thanks for the pointers. Always good to learn a bit more about Rd.

This definitely reads better.
Screen Shot 2021-02-25 at 11 13 58 AM

**Comment by dchud** [comment]
Added ticket #35 to address `clean_value` change to `gcr_result`.

**Pull request merged**